cranelift-codegen/meta/src/isa/x86/legalize.rs

```diff
@@ -290,4 +301,84 @@ pub fn define(shared: &mut SharedDefinitions, x86_instructions: &InstructionGrou
     );
 
     group.build_and_add_to(&mut shared.transform_groups);
+
+    let mut narrow = TransformGroupBuilder::new(
+        "x86_narrow",
+        r#"
+    Legalize instructions by narrowing.
+
+    Use x86-specific instructions if needed."#,
+    )
+    .isa("x86")
+    .chain_with(shared.transform_groups.by_name("narrow").id);
+
+    // SIMD
+    let broadcast_immediate_u8 = Literal::constant(uimm8, 0x00);
+    let broadcast_immediate_f64 = Literal::constant(ieee64, 0x00);
+    let lane_one = Literal::constant(uimm8, 0x01);
+    let b = var("b");
+    let c = var("c");
+    let d = var("d");
+
+    // SIMD splat: 8-bits
+    for ty in LaneTypeIterator::new().filter(|t| t.lane_bits() == 8) {
+        let splat_x8x16 = splat.bind_vector(ty, 128 / ty.lane_bits());
+        let bitcast_f64_to_x8x16 = bitcast.bind_vector(ty, 128 / ty.lane_bits()).bind(F64);
```

Comment:
Same

Reply:
Also what does the `_x8x16` part mean?

Reply:
I want to make clear what the types of the bitcast are and I needed some way to represent `{B|I|F}8x16}`. The first `x` is confusing, I guess; what do you suggest?

Reply:
`any8x16` or `a8x16`

Reply:
yeah, that is better 😄